src/formatters/deployResultFormatter.ts

```diff
@@ -135,11 +135,22 @@ export class DeployResultFormatter extends ResultFormatter {
         failures.push(...fileResponses);
       }
 
-      if (this.result?.response?.details?.componentFailures) {
-        const deployMessages = toArray(this.result.response.details.componentFailures);
+      const deployMessages = toArray(this.result?.response?.details?.componentFailures);
+      if (deployMessages.length > failures.length) {
+        // if there's additional failures in the API response, find the failure and add it to the output
         deployMessages.map((deployMessage) => {
```

Comment:
this could get really loopy on a big deployment fail (use of find within a loop where find is doing multiple string.includes)

perf suggestion: iterate failures once to build a keyed Map< type+fullName>, deployMessage>
then, you can iterate the deployMessages once and use deployMessageMap.has( type + fullName)
No would be a good sign to add it
has=true could still compare the messages (if that's actually a source of dupes)

At the very least, do the (do type and name match?) conditional first so that it **mostly** exits before having to do the string.includes ops